### ISSUE TITLE
access to camera  ?

### ISSUE THREAD

**Issue by weepy** [common-tater/wkwebview-getusermedia-shim#5]
Does this shim give you access to camera and microphone from WkWebview ?

Brill if it does!

**Comment by jessetane** [comment]
Just microphone, but didn't Apple ship GetUserMedia for Safari already?

**Comment by weepy** [comment]
yes - but it's on in safari - not wkwebview.

I saw that cordova were hacking it by passing the camera output base64 encoded into jsland!

**Comment by jessetane** [comment]
Yeah, that's the idea here as well. The project I was working on at the time only needed audio, but I did stub video out, probably wouldn't be too bad to fork and add it yourself:

https://github.com/common-tater/wkwebview-getusermedia-shim/blob/master/WKWebViewGetUserMediaShim/WKWebViewGetUserMediaShim.m#L364

**Comment by weepy** [comment]
Neat - that's pretty clever :-)  Hopefully the perf won't be too bad :)

**Comment by szoszig** [comment]
Hello guys!
I know I'm a bit late, but could you (@weepy ) finally implement the video transfer part (the audio works like a charm)?
I get the frames from camera on iOS, and forward it to js (like you with audio), but in js I am stuck.
Any hint could be useful!
Thx

**Comment by Worien** [comment]
I've use this shim, but for me it's always show the message - "TypeError: undefined is not an object (evaluating 'navigator.mediaDevices.getUserMedia')" 

Also, I've tried to modify the js file and add next lines:
(function (exports) {
  exports.navigator.getUserMedia = getUserMedia
  exports.navigator.mediaDevices.getUserMedia = getUserMedia

but still can't getUserMedia

**Comment by jessetane** [comment]
I can't support this module right now unfortunately

**Comment by Worien** [comment]
Yes, I understand, but maybe you have some ideas why getUserMedia throw the error, instead off redirect call to native code?

**Comment by Worien** [comment]
@szoszig could you please share the code where you get frames from IOS in Objective C?

**Comment by Worien** [comment]
@jessetane can you update the project and make it work for WKWebView in IOS 11-12 and implement the camera part? Of course for some money. If you are ready please contact me via skype - pavel-pugach
